RecoLocalCalo/EcalDeadChannelRecoveryAlgos/src/EcalDeadChannelRecoveryBDTG.cc

```diff
@@ -0,0 +1,160 @@
+//  Original Authors:  S. Taroni, N. Marinelli
+//  University of Notre Dame - US
+//  Created:
+//
+//
+//
+
+#include "RecoLocalCalo/EcalDeadChannelRecoveryAlgos/interface/EcalDeadChannelRecoveryBDTG.h"
+#include "RecoEcal/EgammaCoreTools/interface/EcalClusterTools.h"  // can I use a egammatools here?
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include <iostream>
+#include <TMath.h>
+
+#include <iostream>
+#include <ostream>
+#include <string>
+#include <fstream>
+
+template <>
+void EcalDeadChannelRecoveryBDTG<EBDetId>::loadFile() {
+  readerNoCrack = std::make_unique<TMVA::Reader>("!Color:!Silent");
+
+  for (int i = 0; i < 4; ++i)
+    readerNoCrack->AddVariable(TString("E") + (i + 1) + "/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx_.rEn[i]));
+  for (int i = 5; i < 9; ++i)
+    readerNoCrack->AddVariable(TString("E") + (i + 1) + "/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx_.rEn[i]));
+  readerNoCrack->AddVariable("E1+E2+E3+E4+E6+E7+E8+E9", &(mx_.sumE8));
+  for (int i = 0; i < 9; ++i)
+    readerNoCrack->AddVariable(TString("iEta") + (i + 1), &(mx_.ieta[i]));
+  for (int i = 0; i < 9; ++i)
+    readerNoCrack->AddVariable(TString("iPhi") + (i + 1), &(mx_.iphi[i]));
+
+  readerCrack = std::make_unique<TMVA::Reader>("!Color:!Silent");
+
+  for (int i = 0; i < 4; ++i)
+    readerCrack->AddVariable(TString("E") + (i + 1) + "/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx_.rEn[i]));
+  for (int i = 5; i < 9; ++i)
+    readerCrack->AddVariable(TString("E") + (i + 1) + "/(E1+E2+E3+E4+E6+E7+E8+E9)", &(mx_.rEn[i]));
+  readerCrack->AddVariable("E1+E2+E3+E4+E6+E7+E8+E9", &(mx_.sumE8));
+  for (int i = 0; i < 9; ++i)
+    readerCrack->AddVariable(TString("iEta") + (i + 1), &(mx_.ieta[i]));
+  for (int i = 0; i < 9; ++i)
+    readerCrack->AddVariable(TString("iPhi") + (i + 1), &(mx_.iphi[i]));
+
+  reco::details::loadTMVAWeights(readerNoCrack.get(), "BDTG", bdtWeightFileNoCracks_.fullPath());
+  reco::details::loadTMVAWeights(readerCrack.get(), "BDTG", bdtWeightFileNoCracks_.fullPath());
+}
+
+template <typename T>
+EcalDeadChannelRecoveryBDTG<T>::EcalDeadChannelRecoveryBDTG() {}
+
+template <typename T>
+EcalDeadChannelRecoveryBDTG<T>::~EcalDeadChannelRecoveryBDTG() {}
+
+template <>
+void EcalDeadChannelRecoveryBDTG<EBDetId>::setParameters(const edm::ParameterSet &ps) {
+  bdtWeightFileNoCracks_ = ps.getParameter<edm::FileInPath>("bdtWeightFileNoCracks");
+  bdtWeightFileCracks_ = ps.getParameter<edm::FileInPath>("bdtWeightFileCracks");
+  this->loadFile();
```

Comment:
it looks like the TMVA initialization is unconditional.
Please add a check of the value of recoverEBIsolatedChannels and call loadFile only if it is set.
(this will avoid having this code essentially enabled in production).